example/cypress/integration/input.spec.js

```diff
@@ -5,7 +5,7 @@ describe('Attach file to an input element', () => {
 
   it('successfully uploads a file', () => {
     cy.fixture('cy.png', 'base64').then(fileContent => {
-      cy.get('[data-cy="input"]').upload(
+      cy.get('[data-cy="input"]').first().upload(
```

Comment:
I suggest adding some extra data attributes to distinguish between input and hidden-input:
like `data-cy="input-container"` and `data-cy="hidden-input-container"`. This will make the selector more specific and straightforward, so the real position in DOM will not really matter.